File: src/services/tauri/skins.ts
import type { SkinInfo } from "../../types";
import { invokeTyped } from "./core";

export function getActiveSkin(): Promise<SkinInfo | null> {
  return invokeTyped("get_active_skin");
}

export function setActiveSkinBase64(name: string, model: string, data: string): Promise<SkinInfo> {
  return invokeTyped("set_active_skin_base64", { name, model, data });
}

export function setActiveSkinUrl(
  url: string,
  name: string | undefined,
  model: string
): Promise<SkinInfo> {
  return invokeTyped("set_active_skin_url", { url, name, model });
}

export function clearActiveSkin(): Promise<void> {
  return invokeTyped("clear_active_skin");
}

export function getActiveCape(): Promise<string | null> {
  return invokeTyped("get_active_cape");
}

export function setActiveCapeBase64(data: string): Promise<string> {
  return invokeTyped("set_active_cape_base64", { data });
}

export function setActiveCapeUrl(url: string): Promise<string> {
  return invokeTyped("set_active_cape_url", { url });
}

export function clearActiveCape(): Promise<void> {
  return invokeTyped("clear_active_cape");
}

File: src-tauri/src/commands/skins.rs
use super::map_app_result;
use crate::models::SkinInfo;

#[tauri::command]
pub async fn get_active_skin(app: tauri::AppHandle) -> Result<Option<SkinInfo>, String> {
    map_app_result(crate::skins::get_active_skin(app).await)
}

#[tauri::command]
pub async fn set_active_skin_base64(
    app: tauri::AppHandle,
    name: String,
    model: String,
    data: String,
) -> Result<SkinInfo, String> {
    map_app_result(crate::skins::set_active_skin_base64(app, name, model, data).await)
}

#[tauri::command]
pub async fn set_active_skin_url(
    app: tauri::AppHandle,
    url: String,
    name: Option<String>,
    model: String,
) -> Result<SkinInfo, String> {
    map_app_result(crate::skins::set_active_skin_url(app, url, name, model).await)
}

#[tauri::command]
pub async fn clear_active_skin(app: tauri::AppHandle) -> Result<(), String> {
    map_app_result(crate::skins::clear_active_skin(app).await)
}

#[tauri::command]
pub async fn get_active_cape(app: tauri::AppHandle) -> Result<Option<String>, String> {
    map_app_result(crate::skins::get_active_cape(app).await)
}

#[tauri::command]
pub async fn set_active_cape_base64(app: tauri::AppHandle, data: String) -> Result<String, String> {
    map_app_result(crate::skins::set_active_cape_base64(app, data).await)
}

#[tauri::command]
pub async fn set_active_cape_url(app: tauri::AppHandle, url: String) -> Result<String, String> {
    map_app_result(crate::skins::set_active_cape_url(app, url).await)
}

#[tauri::command]
pub async fn clear_active_cape(app: tauri::AppHandle) -> Result<(), String> {
    map_app_result(crate::skins::clear_active_cape(app).await)
}
